tests/geomodel/core/test-geomodel-build.cpp

```diff
@@ -0,0 +1,641 @@
+    }
+
+    // Create the Layers
+    for( index_t layer = 0; layer < nb_layers; layer++ )
+    {
+        create_geology_entity( Layer3D::type_name_static(), builder );
+        /*builder.geology.create_geological_entity( Layer3D::type_name_static()
+         * );*/
+    }
+
+    // Then we create the GeoModelMeshEntity
+    // Create the Corners
+    for( index_t corner = 0; corner < nb_corners; corner++ )
+    {
+        create_topology_entity( Corner3D::type_name_static(), builder );
+        /*   builder.topology.create_mesh_entity( Corner3D::type_name_static()
+         * );*/
+    }
+
+    // Create the Lines
+    for( index_t lines = 0; lines < nb_lines; lines++ )
+    {
+        create_topology_entity( Line3D::type_name_static(), builder );
+        /* builder.topology.create_mesh_entity( Line3D::type_name_static() );*/
+    }
+
+    // Create the Surfaces
+    for( index_t surface = 0; surface < nb_surfaces; surface++ )
+    {
+        create_topology_entity( Surface3D::type_name_static(), builder );
+        /* builder.topology.create_mesh_entity( Surface3D::type_name_static()
+         * );*/
+    }
+
+    // Create the Regions
+    for( index_t region = 0; region < nb_regions; region++ )
+    {
+        create_topology_entity( Region3D::type_name_static(), builder );
+        /* builder.topology.create_mesh_entity( Region3D::type_name_static()
+         * );*/
+    }
+
+    //#############################
+    // Setting the Geometry       #
+    //#############################
+
+    // We declare the coordinates of the corners. We arrange the corner in a
+    // table
+    vec3 corners_table[nb_corners];
+    corners_table[0] = vec3( 0, 0, 0 );
+    corners_table[1] = vec3( 4, 0, 0 );
+    corners_table[2] = vec3( 2, 2, 0 );
+    corners_table[3] = vec3( 2, 1, 4 );
+    corners_table[4] = vec3( 2, 1, -4 );
+
+    // We associate the coordinates with the corners
+    for( index_t corner = 0; corner < nb_corners; ++corner )
+    {
+        builder.geometry.set_corner( corner, corners_table[corner] );
+    }
+
+    // We associate the coordinates with the lines
+    // We create a vector cur_coor_line containing the 2 vertices
+    // for each line. Of course, you can have more vertices in a Line
+    std::vector< vec3 > cur_coor_line( 2 );
+    cur_coor_line[0] = corners_table[0];
+    cur_coor_line[1] = corners_table[3];
+    builder.geometry.set_line( 0, cur_coor_line );
+
+    const GeoModelMeshEntity3D& mesh_entity1 =
+        geomodel.mesh_entity( line_type_name_static(), 0 );
+
+    // being paranoid
+    const Line3D* line = dynamic_cast< const Line3D* >( &mesh_entity1 );
+    if( line == nullptr )
+    {
+        throw RINGMeshException( "RINGMesh Test", "Mesh entity is not a Line" );
+    }
+
+    if( line->vertex( 0 ) != cur_coor_line[0]
+        || line->vertex( 1 ) != cur_coor_line[1] )
+    {
+        throw RINGMeshException(
+            "RINGMesh Test", "The line's corners are not equal" );
```

Comment:
You are building the model and testing the model building. This code nlock should be un test_geomodel_build, in my opinion.